spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -1108,6 +1108,9 @@ trait OptimisticTransactionImpl extends TransactionalWrite
       val readRowIdHighWatermark =
         RowId.extractHighWatermark(snapshot).getOrElse(RowId.MISSING_HIGH_WATER_MARK)
 
+      val tagsWithPreservingRowTrackingIfEnabled =
+        RowTracking.addPreservedRowTrackingTagIfNotSet(snapshot, tags)
```

Comment:
You should set this tag in DELETE, other operations aren't preserving row tracking information.

Alternatively, set it to true by default here but explicitly set the tag to false in operations that don't preserve row tracking information: UPDATE, MERGE, OPTIMIZE, ..?

Reply:
@johanl-db Done!